### ISSUE TITLE
scrollToItem + changing number of items sometimes glitches

### ISSUE THREAD

**Issue by sk1p** [bvaughn/react-window#45]
I'm writing a file listing widget using `FixedSizeList`, and after changing to a different folder, I want to scroll back to the top. This sometimes causes the bottom half of all items to disappear, see this codesandbox:

https://8zm2lzr10j.codesandbox.io/

I noticed that in the broken state, there is a negative `scrollOffset`. That's about as far as I got with debugging.

Tested with `react-window` 1.1.1 on Chrome 68.0.3440.106 on Linux, Firefox 60.1.0esr (64-bit) on Linux.

**Comment by bvaughn** [comment]
Hm... I haven't had time to look into this really, but your app works if you delay the `scrollToItem` call until after state has been updated, e.g.:
```js
this.setState(
  { numItems: 32 }, // Update the itemCount
  () => listRef.current.scrollToItem(0) // Once the List count has updated, scroll to the new item
);

I'll have to dig in more and determine if this is a bug or not. Seems likely, I guess. Thanks for the repro. :smile:

**Comment by bvaughn** [comment]
Looks like a bug in my [offset calculation](https://github.com/bvaughn/react-window/blob/86c24c07ef7bde61b920692ee361d743e0efee60/src/FixedSizeList.js#L24-L27) that doesn't properly account for a list that contains too few items to even fill the visible window.

**Comment by bvaughn** [comment]
Bugfix published in v1.1.2

**Comment by sk1p** [comment]
Thanks for the quick fix, and enjoy the rest of your vacation!
